compiler/ast.nim

```diff
@@ -991,6 +990,24 @@ type
   TImplication* = enum
     impUnknown, impNo, impYes
 
+template nodeId(n: PNode): int = cast[int](n)
+
+type Gconfig = object
+  # we put comments in a side channel to avoid increasing `sizeof(TNode)`, which
+  # reduces memory usage given that `PNode` is the most allocated type by far.
+  comments: Table[int, string] # nodeId => comment
+
+var gconfig {.threadvar.}: Gconfig
+
+proc comment*(n: PNode): string =
+  gconfig.comments.getOrDefault(n.nodeId)
```

Comment:
Educated guess: if you add another node flag you can skip the getOrDefault step in 99% of all cases and bring back the speed of the old approach.

Reply:
good idea, I've now implement this and it works, giving same memory improvement without incurring a cost; 
however there was 1 subtlety but it all works fine, see comments in the PR; TLDR: there's a small amount of leak in the comments table that is entirely justified for performance reasons; future work can improve this if needed, but it probably won't be needed.